lib/puppet/transaction/report.rb

```diff
@@ -176,7 +178,7 @@ def add_resource_status(status)
 
   # @api private
   def compute_status(resource_metrics, change_metric)
-    if resources_failed_to_generate || (resource_metrics["failed"] || 0) > 0
+    if resources_failed_to_generate || (resource_metrics["failed"] || 0) > 0 || @logs.any? { |lg| lg.level == :err }
```

Comment:
So if you're making this change (which I guess is how we calculate status), being called from `finalize_report` at https://github.com/hlindberg/puppet/blob/36225c3c81c0f7a03f46d86802c6060814881d7f/lib/puppet/transaction/report.rb#L201

Then I don't believe the above changes to `def <<(msg)` are necessary, are they?